### PR TITLE
Issue 103 lead acid LOQS model

### LINKED ISSUE #103: Implement lead-acid LOQS model
Leading-order quasi-static model for lead-acid: a system of ODEs for concentration and porosity. The voltage is calculated afterwards. 

Requires making `standard_parameters_lead_acid.py`

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Mainly fixes #103 (LOQS model), but made a few other changes:
- added a `symbol.evaluates_to_scalar()` node (for broadcasting), could be useful for #54 
- added `standard_parameters_lead_acid` for lead-acid parameters
- some changes around how a model is created and discretised, unifying process for initial conditions and rhs 
- added a simple ODE model to test functionality not captured by `ReactionDiffusionModel`

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pybamm/discretisations/base_discretisation.py [hunk 1]
```diff
@@ -533,11 +508,13 @@ def check_model(self, model):
...
         # Check variables in variable list against rhs
+        # Be lenient with size check if the variable in model.variables is broadcasted
         for var in model.rhs.keys():
             if var.name in model.variables.keys():
-                assert (
-                    model.rhs[var].evaluate(0, y0).shape
-                    == model.variables[var.name].evaluate(0, y0).shape
+                assert model.rhs[var].evaluate(0, y0).shape == model.variables[
+                    var.name
+                ].evaluate(0, y0).shape or isinstance(
```
Comment (martinjrobins) [inline pybamm/discretisations/base_discretisation.py]: why is this necessary? shouldn't the discretisation replace all the Broadcast nodes with NumpyBroadcast?

File: pybamm/expression_tree/unary_operators.py [hunk 2]
```diff
@@ -164,22 +164,35 @@ class NumpyBroadcast(Broadcast):
...
+        # if child is a state vector, check that it has the right shape and then
+        # transpose, broadcast, transpose again
+        # QUESTION: is there a better way of doing this than transposing twice?
+        elif isinstance(child, pybamm.StateVector):
+            assert child_eval.shape[0] == 1, ValueError(
+                """child_eval should have shape (1,n), not {}""".format(
+                    child_eval.shape
+                )
             )
+            return (child_eval.T * self.broadcasting_vector).T
```
Comment (martinjrobins) [inline pybamm/expression_tree/unary_operators.py]: could use `np.repeat(child_eval,broadcasting_vector_size,axis=0)`?

File: pybamm/models/lead_acid/full.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
+#
+# Lead-acid full (1D) model
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+
+# import pybamm
```
Comment (martinjrobins) [inline pybamm/models/lead_acid/full.py]: should this (empty) file be here?

Comment (tinosulzer) [reply pybamm/models/lead_acid/full.py]: nope, just got lazy with branching, removing

File: pybamm/discretisations/base_discretisation.py (hunk 1 above)
Comment (tinosulzer) [reply pybamm/discretisations/base_discretisation.py]: changing to NumpyBroadcast

File: pybamm/expression_tree/unary_operators.py (hunk 2 above)
Comment (tinosulzer) [reply pybamm/expression_tree/unary_operators.py]: Cool, this is a bit faster and clearer so have updated. This `evaluate` still feels a bit hacky but will just improve as and when needed
